hw_28_students/js/script.js

```diff
@@ -0,0 +1,200 @@
+// У кожного студента повинні бути такі методи
+// getAverageMark() - повертає середню оцінку
+// getMinMark() - повертає найменшу оцінку
+// getMaxMark() - повертає максимальну оцінку
+
+class Student extends Human {
+    constructor({ fName, lName, age, marks }) {
+        super({ fName, lName, age } );
+        this.marks = marks;
+    }
+
+    getAverageMark() {
+        return this.marks.reduce((total, mark) => (total + mark)) / this.marks.length;
+    }
+
+    getMinMark() {
+        return Math.min(...this.marks);
+    }
+
+    getMaxMark() {
+        return Math.max(...this.marks);
+    }
+}
+
+// 2.1 Створити FakeStudent
+
+class FakeStudent extends Student {
+    #cheatedmarks = this.#cheat(this.marks);
+
+    constructor({ fName, lName, age, marks }) {
+        super({ fName, lName, age, marks });
+    }
+
+    #cheat(values) {
+        return values.map(value => value * 2 > maxMark ? value = maxMark : value *= 2 ) 
+    }
+
+    getAverageMark() {
+        return this.#cheatedmarks.reduce((total, mark) => (total + mark)) / this.#cheatedmarks.length;
+    }
+
+    getMinMark() {
+        return Math.min(...this.#cheatedmarks);
+    }
+
+    getMaxMark() {
+        return Math.max(...this.#cheatedmarks);
+    }
+
+}
+
+// 3. Створити клас Teacher який приймає в аргументи об'єкт
+// і створює проперті group(масив не менше 5 - ти студентів створених за допомогою класа Student)
+// та містить методи
+// getListOfNamesByAverageMark() - віддає масив імен студентів відсортований за найвищою середньою оцінкою.
+// getStudentByName(name) - отримати один об'єкт студента за ім'ям.
+// removeStudentByName(name) - видалити об'єкт студента, знайденого за ім'ям.
+// updateStudentByName(new Student(...), name) - знайти об'єкт студента по name та замінити на student (новий екземпляр класа Student)
+
+// 3.1 Створити метод, що шукає fakeStudent
+
+class Teacher extends Human {
+    constructor({ fName, lName, age, group }) {
+        super({ fName, lName, age });
+        this.group = group;
+    }
+
+    getListOfNamesByAverageMark() {
+        const groupSorted = [...this.group].sort((a, b) => (b.getAverageMark() - a.getAverageMark()));
+        const groupSortedName = groupSorted.map(element => element.getFullName());
+        
+        return groupSortedName;
+    }
+
+    getStudentByName(name) {
+        const result = this.group.filter(element => element.getFullName() === name);
+
+        return result;
+    }
+
+    removeStudentByName(name) {
+        this.group = this.group.filter(element => element.getFullName() != name);
+
+        return this.group;
+    }
+
+    updateStudentByName(newStudent, name) {
+        this.removeStudentByName(name);
+        this.group.push(newStudent);
+
+        return this.group;
+    }
+
+    findFakeStudent() {
+        const fakeStudent = this.group.filter(element => element.getMinMark() != Math.min(...element.marks));
```

Comment:
same

Reply:
also if you wanna find just one element in array, it's better to use array.find()